src/main/kotlin/StringAddCalculator.kt

```diff
@@ -0,0 +1,28 @@
+class StringAddCalculator {
+    fun add(text: String?): Int {
+        text ?: return 0
+        if (text.isEmpty()) {
+            return 0
+        }
+
+        val numbers = splitWithDelimiter(text)
+
+        return numbers.sumOf { toPositiveInt(it) }
+    }
+
+    private fun splitWithDelimiter(text: String): List<String> {
+        if (text.matches(Regex("//(.)\n(.*)"))) {
+            val customDelimiter = text[2]
+            return text.substring(4).split(customDelimiter)
+        }
+        return text.split("[,|:]".toRegex())
+    }
+
+    private fun toPositiveInt(text: String): Int {
+        val number = text.toInt()
+        if (number < 0) {
+            throw RuntimeException()
+        }
```

Comment:
```suggestion
        require(number >= 0) { "음수는 입력될 수 없습니다." }
```
코틀린에서는 [require](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html)와 [check](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/check.html)를 제공하고 있어 활용한다면 좀 더 코틀린스러운 코드를 작성할 수 있습니다.

예외가 발생하는 것까지가 힌트사항이나 적절한 예외 메시지도 추가되면 좋을 것 같아요.

Reply:
아하 감사합니다! 🙇